### PR TITLE
add cloud-config and some more fixes

### PR DESCRIPTION
**Pull request by saikat-royc**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
1. Add support for cloud-config for GCE cloud provider (2bd88cd) to enable `--cloud-config=/etc/gce.conf ` args for managed GKE driver
2. Remove dependency on nodeId for controller service (c78f30c)
3. Minor fixes: a. kustomize version b. fix do-driver-build (there was a bug which did not replace the image tag) (4b18df2)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Enable support for cloud provider config
```

### CODE REVIEW

File: pkg/cloud_provider/cloud.go [hunk 1]
```diff
@@ -17,40 +17,160 @@ limitations under the License.
...
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
 	"sigs.k8s.io/gcp-filestore-csi-driver/pkg/cloud_provider/file"
-	"sigs.k8s.io/gcp-filestore-csi-driver/pkg/cloud_provider/metadata"
 )
 
 type Cloud struct {
 	File file.Service
-	Meta metadata.Service
+	//Meta    metadata.Service
```
Comment (mattcary) [inline pkg/cloud_provider/cloud.go:39]: Remove commented out Meta?

**Review by mattcary — COMMENTED** [review]
Maybe also add a line in the deployment documentation explaining how the different options work?

Also I'm not sure we need hostNetwork for the filestore driver---there may be enough of the required info from the WI metadata server (?). That's probably out of scope of this PR, though.

**Comment by mattcary** [comment]
> Also I'm not sure we need hostNetwork for the filestore driver---there may be enough of the required info from the WI metadata server (?). That's probably out of scope of this PR, though.

It would be nice to document as well that hostNetwork means that WI can't be used to authorize the driver, the token source needs to be used.

**Comment by saikat-royc** [comment]
> It would be nice to document as well that hostNetwork means that WI can't be used to authorize the driver, the token source needs to be used.

Ack. I ll add the doc changes in a separate PR, as this one is too big already

File: pkg/cloud_provider/cloud.go (hunk 1 above)
Comment (saikat-royc) [reply pkg/cloud_provider/cloud.go:39]: Done

**Comment by saikat-royc** [comment]
> > Also I'm not sure we need hostNetwork for the filestore driver---there may be enough of the required info from the WI metadata server (?). That's probably out of scope of this PR, though.
> 
> It would be nice to document as well that hostNetwork means that WI can't be used to authorize the driver, the token source needs to be used.

Created an internal bug to investigate hostNetwork dependency and update the readme

**Pull request merged**